src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java

```diff
@@ -0,0 +1,158 @@
+   *     EST is 240 minutes behind UTC.
+   */
+  // TODO: Create a timeSlotsForWeek method.
+  public static List<AvailabilityTimeSlot> timeSlotsForDay(Instant instant, String timezoneOffset) {
+    ZonedDateTime day = generateDay(instant, timezoneOffset);
+    ZoneId zoneId = day.getZone();
+    String dayOfWeek = day.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.US);
+    int year = day.getYear();
+    int month = day.getMonthValue();
+    int dayOfMonth = day.getDayOfMonth();
+
+    String date = generateDate(dayOfWeek, month, dayOfMonth);
+    List<String> times = generateTimes();
+    List<String> utcEncodings = generateUTCEncodings(year, month, dayOfMonth, zoneId);
+    List<Boolean> selectedStatuses = getSelectedStatuses();
+    return generateTimeSlots(utcEncodings, times, date, selectedStatuses);
+  }
+
+  // Uses an Instant and a timezoneOffset String to create a ZonedDateTime instance
+  // for the day specified by the Instant.
+  private static ZonedDateTime generateDay(Instant instant, String timezoneOffset) {
+    ZoneOffset timeZoneOffset = convertStringToOffset(timezoneOffset);
+    ZoneId zoneId = ZoneId.ofOffset("UTC", timeZoneOffset);
+    return instant.atZone(zoneId);
+  }
+
+  // This method takes the timezoneOffset String and converts it
+  // into a proper ZoneOffset instance.
+  private static ZoneOffset convertStringToOffset(String timezoneOffset) {
+    int offsetTotalMinutes = Integer.parseInt(timezoneOffset);
+    int offsetHours = offsetTotalMinutes / 60;
+    int offsetMinutes = offsetTotalMinutes % 60;
+    return ZoneOffset.ofHoursMinutes(offsetHours, offsetMinutes);
+  }
+
+  // Returns a readable date string such as "Tue 7/7".
+  private static String generateDate(String dayOfWeek, int month, int dayOfMonth) {
+    return dayOfWeek + " " + month + "/" + dayOfMonth;
+  }
+
+  // This method returns a list of UTC Strings for a single day, creating one String per Time
+  // Slot.
+  private static List<String> generateUTCEncodings(
+      int year, int month, int dayOfMonth, ZoneId zoneId) {
+    List<String> utcEncodings = new ArrayList<String>();
+    int[] minutes = {0, 15, 30, 45};
+    for (int i = TIME_8AM; i <= TIME_7PM; i++) {
+      for (int j = 0; j < 4; j++) {
+        LocalDateTime localTimeSlot = LocalDateTime.of(year, month, dayOfMonth, i, minutes[j]);
+        ZonedDateTime utcTimeSlot =
+            ZonedDateTime.of(localTimeSlot, zoneId).withZoneSameInstant(ZoneOffset.UTC);
+        String formattedUTCTimeSlot = utcTimeSlot.format(DateTimeFormatter.ISO_INSTANT);
+        utcEncodings.add(formattedUTCTimeSlot);
+      }
+    }
+    return utcEncodings;
+  }
+
+  // TODO: Access the time slots from data store to tell if they are selected or not.
+  // This method will tell whether or not a time slot has already been selected. (See
+  // TODO above).
+  private static List<Boolean> getSelectedStatuses() {
```

Comment:
This function almost certainly needs to accept the list of times (if not for now, than for later).

Reply:
Done